### PR TITLE
test: migrate s3 extension tests to JUnit 5

### PR DESCRIPTION
**Pull request by FrankChen021**
## Description

Migrates the `extensions-core/s3-extensions` test suite from JUnit 4 to JUnit 5.

This is Batch 6 of the extensions-core migration and is part of [#13948](https://github.com/apache/druid/issues/13948).

Depends on the earlier migration batches in [#19875](https://github.com/apache/druid/pull/19875), [#19876](https://github.com/apache/druid/pull/19876), [#19877](https://github.com/apache/druid/pull/19877), [#19878](https://github.com/apache/druid/pull/19878), and [#19879](https://github.com/apache/druid/pull/19879).

The migration replaces JUnit 4 assertions, rules, and lifecycle methods with Jupiter equivalents, including `@TempDir` and `assertThrows`.

## Tests

```text
mvn -ntp test -pl extensions-core/s3-extensions -am \
  -Dtest='org.apache.druid.catalog.model.table.S3InputSourceDefnTest,org.apache.druid.data.input.s3.**,org.apache.druid.storage.s3.**' \
  -Dsurefire.failIfNoSpecifiedTests=false \
  -Pskip-static-checks -Dweb.console.skip=true -T1C
```

Result: 235 tests passed; 0 failures, errors, or skips.

### CODE REVIEW

File: extensions-core/s3-extensions/src/test/java/org/apache/druid/storage/s3/S3DataSegmentArchiverTest.java [hunk 1]
```diff
@@ -100,7 +99,7 @@ public String getArchiveBaseKey()
       .size(0)
       .build();
 
-  @BeforeClass
+
   public static void setUpStatic()
```
Comment (FrankChen021) [inline extensions-core/s3-extensions/src/test/java/org/apache/druid/storage/s3/S3DataSegmentArchiverTest.java:104]: [P2] Restore the class-level setup hook

The former `@BeforeClass` annotation was removed without adding `@BeforeAll`, so `setUpStatic()` never initializes the pusher bucket and base key. Both archive tests override `move()` and ignore its target load spec, allowing the invalid empty configuration to pass silently. Add `@BeforeAll` or initialize `PUSHER_CONFIG` statically.

**Review by FrankChen021 — COMMENTED** [review]
| Severity | Findings |
| --- | ---: |
| P0 | 0 |
| P1 | 0 |
| P2 | 1 |
| P3 | 0 |
| Total | 1 |

| Severity | Findings |
| --- | ---: |
| P0 | 0 |
| P1 | 0 |
| P2 | 1 |
| P3 | 0 |
| Total | 1 |

Reviewed 27 of 27 changed files.

---

This is an automated review by Codex GPT-5.6-Sol

File: extensions-core/s3-extensions/src/test/java/org/apache/druid/storage/s3/S3DataSegmentArchiverTest.java (hunk 1 above)
Comment (FrankChen021) [reply extensions-core/s3-extensions/src/test/java/org/apache/druid/storage/s3/S3DataSegmentArchiverTest.java:104]: Fixed in 7f63dfa376: setUpStatic now has @BeforeAll. S3DataSegmentArchiverTest passed all 4 tests.

File: extensions-core/s3-extensions/src/test/java/org/apache/druid/storage/s3/S3DataSegmentMoverTest.java [hunk 2]
```diff
@@ -211,30 +214,33 @@
...
+                "baseKey/test/2013-01-01T00:00:00.000Z_2013-01-02T00:00:00.000Z/1/0/index.zip",
+                "bucket",
+                "DOES NOT EXIST"
+            ),
+            ImmutableList.of("dim1", "dim1"),
+            ImmutableList.of("metric1", "metric2"),
+            NoneShardSpec.instance(),
+            0,
+            1
+        ), ImmutableMap.of("bucket", "DOES NOT EXIST", "baseKey", "baseKey2"))
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-core/s3-extensions/src/test/java/org/apache/druid/storage/s3/S3DataSegmentMoverTest.java:242]: Evaluated as baseline: this deprecated DataSegment constructor already existed on master before this JUnit migration. This PR does not alter it, so deprecated API cleanup is separate scope and is left unchanged.

File: extensions-core/s3-extensions/src/test/java/org/apache/druid/storage/s3/ServerSideEncryptingAmazonS3Test.java [hunk 3]
```diff
@@ -233,13 +236,13 @@
...
 
-    Assert.assertNull("TransferManager should be null when disabled", transferManager);
+    Assertions.assertNull(transferManager, "TransferManager should be null when disabled");
   }
 
   @Test
   public void testBuilder() throws IOException
   {
-    File testFile = temporaryFolder.newFile("test-builder.txt");
+    File testFile = File.createTempFile("test-builder", ".txt", temporaryFolder);
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-core/s3-extensions/src/test/java/org/apache/druid/storage/s3/ServerSideEncryptingAmazonS3Test.java]: Fixed in 7f63dfa376 by removing the unused testFile local. The two focused S3 test classes passed all 14 tests.

**Review by FrankChen021 — COMMENTED** [review]
I have reviewed the updated code for correctness, edge cases, concurrency, and integration risks; no issues found.

Reviewed 27 of 27 changed files. The prior lifecycle finding is resolved: `setUpStatic()` now runs under `@BeforeAll`; the other incremental cleanup is safe.

---

This is an automated review by Codex GPT-5.6-Sol

**Review by GWphua — APPROVED** [review]
